sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -96,6 +97,8 @@ private[spark] case class SparkUserDefinedFunction(
     nullable: Boolean = true,
     deterministic: Boolean = true) extends UserDefinedFunction {
 
+  CharVarcharUtils.failIfHasCharVarchar(dataType)
```

Comment:
do we need it? We already have the check in `UDFRegistration`

Reply:
Could this happen when one person defines the UDF but another guy tries to register and use it?